### ISSUE TITLE
Highlighting only first error/warning

### ISSUE THREAD

**Issue by AlefCS** [dense-analysis/ale#3656]
## Information

**VIM version**

VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Feb 21 2021 21:02:22)
Correções incluídas: 1-2539

Operating System: Ubuntu 18.04.5 LTS

## What went wrong

With C++ files (didn't test any other) ALE is highlighting only the first error/warning. But if I run `:ALEInfo` I can see that all errors/warnings are detected.

## Reproducing the bug

1. Opened a `.cpp` file.
2. Put some code with multiple erros/warnings as this below:
```
#include <s.h>

int main() {
    printf("alef");
    int a = 0;
    float b = 15.3;
    a = b
    return 0;
}
```

### :ALEInfo
```
 Current Filetype: cpp
Available Linters: ['cc', 'ccls', 'clangcheck', 'clangd', 'clangtidy', 'clazy', 'cppcheck', 'cpplint', 'cquery', 'flawfinder']
   Linter Aliases:
'cc' -> ['gcc', 'clang', 'g++', 'clang++']
  Enabled Linters: ['cc', 'ccls', 'clangcheck', 'clangd', 'clangtidy', 'clazy', 'cppcheck', 'cpplint', 'cquery', 'flawfinder']
  Ignored Linters: []
 Suggested Fixers: 
  'astyle' - Fix C/C++ with astyle.
  'clang-format' - Fix C/C++ and cuda files with clang-format.
  'clangtidy' - Fix C/C++ and ObjectiveC files with clang-tidy.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
  'uncrustify' - Fix C, C++, C#, ObjectiveC, ObjectiveC++, D, Java, Pawn, and VALA files with uncrustify.
 Linter Variables:

let g:ale_cpp_cc_executable = '<auto>'
let g:ale_cpp_cc_options = '-std=c++14 -Wall'
let g:ale_cpp_ccls_executable = 'ccls'
let g:ale_cpp_ccls_init_options = {}
let g:ale_cpp_clangcheck_executable = 'clang-check'
let g:ale_cpp_clangcheck_options = ''
let g:ale_cpp_clangd_executable = 'clangd'
let g:ale_cpp_clangd_options = ''
let g:ale_cpp_clangtidy_checks = []
let g:ale_cpp_clangtidy_executable = 'clang-tidy'
let g:ale_cpp_clangtidy_extra_options = ''
let g:ale_cpp_clangtidy_options = ''
let g:ale_cpp_clazy_checks = ['level1']
let g:ale_cpp_clazy_executable = 'clazy-standalone'
let g:ale_cpp_clazy_options = ''
let g:ale_cpp_cppcheck_executable = 'cppcheck'
let g:ale_cpp_cppcheck_options = '--enable=style'
let g:ale_cpp_cpplint_executable = 'cpplint'
let g:ale_cpp_cpplint_options = ''
let g:ale_cpp_cquery_cache_directory = '/home/alef.lds/.cache/cquery'
let g:ale_cpp_cquery_executable = 'cquery'
let g:ale_cpp_flawfinder_executable = 'flawfinder'
let g:ale_cpp_flawfinder_minlevel = 1
let g:ale_cpp_flawfinder_options = ''
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_disable_lsp = 0
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {}
let g:ale_linters_explicit = 0
let g:ale_linters_ignore = {}
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_root = {}
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - failure) clang++
(executable check - success) gcc
(finished - exit code 1) ['/usr/bin/zsh', '-c', '''gcc'' -S -x c++ -o /dev/null -iquote ''/tmp'' -std=c++14 -Wall - < ''/tmp/vGkkDyX/1/teste.cpp''']

<<<OUTPUT STARTS>>>
<stdin>:1:10: fatal error: s.h: Arquivo ou diretório inexistente
compilation terminated.
<<<OUTPUT ENDS>>>

(executable check - failure) clang-check
(executable check - failure) clang-tidy
(executable check - failure) clazy-standalone
(executable check - failure) cppcheck
(executable check - failure) cpplint
(executable check - failure) flawfinder
```

**Comment by hsanson** [comment]
I this the correct ALEInfo output? It shows only one error in the output:

```
<<<OUTPUT STARTS>>>
<stdin>:1:10: fatal error: s.h: Arquivo ou diretório inexistente
compilation terminated.
<<<OUTPUT ENDS>>>
```

**Comment by AlefCS** [comment]
> Is this the correct ALEInfo output? It shows only one error in the output:
> 
> ```
> <<<OUTPUT STARTS>>>
> <stdin>:1:10: fatal error: s.h: Arquivo ou diretório inexistente
> compilation terminated.
> <<<OUTPUT ENDS>>>
> ```

Sorry for that. I saw the history showing several errors in another example and thought that all of them was about this one, where the header does not exist and there is a missing comma in the end of a command.

Did some research and figure it out that problem with the header is a fatal error, so `gcc` will stop right there and no other errors would be showed. So nothing wrong with ALE, I'm closing this issue.

Thank you for making me see what I couldn't.
